nakedret.go

```diff
@@ -34,6 +34,8 @@ func usage() {
 type returnsVisitor struct {
 	f         *token.FileSet
 	maxLength uint
+	root      *returnsVisitor
```

Comment:
Why do we need to pass the full `returnsVisitor` object down? Could we not just define a `found` boolean similar to what has been done for `maxLength`?

Reply:
The problem is that the thing finding the issue needs to tell the main loop that it found an issue, and it currently is not.
We certainy don't need to do it like this: there are (at least :stuck_out_tongue:) two alternative approaches to this feature, neither of which requires this:
1. set a global 'found something' flag when we find something, exit depending on that. This seemed a bit rude. I could write it this way if you asked me to though.
2. refactor everything so that the the main loop becomes e.g.

       found := false
       for somefinding := range checkNakedReturns(...) {
               fmt.Println(...)
               found = true
       }
       if found && setExitStatus { ... etc

   this would bring nakedret closer to your prealloc (if `checkNakedReturns` returned a list, although the above suggests a channel), which might make the maintenance burden lighter for you. I'd be happy to do this refactor if that's your preference.

Reply:
Ah, i see now. Thanks for the explanation. This approach seems reasonable. Thanks again for the PR. :slightly_smiling_face: